server/src/test/java/io/crate/integrationtests/UpdateIntegrationTest.java

```diff
@@ -1111,4 +1107,42 @@ public void test_update_object_array_by_elements_with_column_policies() {
             .isExactlyInstanceOf(ColumnUnknownException.class)
             .hasMessage("Column a['c'] unknown");
     }
+
+    @Test
+    public void test_update_on_table_with_big_refresh_interval_emptied_by_delete_return_0_rows() {
```

Comment:
Isn't the change for the existing test enough, or what are these covering additionally?

Reply:
If it's about `test_update_continues_on_error_if_at_least_1_row_was_updated` - I removed assertion for "no_updates_at_all" since it's already covered by existing `testUpdateSetInvalidGeneratedColumnOnly`

but expression `update test set a = a + 100 * (a % 2) `is partial failure and is a new test case, we don't have tests where `UPDATE` actually affects some rows and fails on some others. AFAIK, all existing UPDATE tests are either all pass/all fail.

If comment was about `test_update_on_table_with_big_refresh_interval_emptied_by_delete_return_0_rows`:
It's basically same with existing in a way that "saw an error, skip it" - but root cause of the error is different (collect reader is stale and unaware about delete results". I guess error doesn't matter here, we care only about continue so can drop this test case?

Reply:
I created dedicated https://github.com/crate/crate/pull/14758 to fix delete + update. It's more pinpoint and can be back ported to 5. whereas this PR is more like a feature. 

Also, another PR is also needed to ensure that implementation of https://github.com/crate/crate/issues/12218  won't expose https://github.com/crate/crate/issues/14707 again.